src/audio/host-zephyr.c

```diff
@@ -878,12 +890,11 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
 	hd->chan->period = config->period;
 
 	memset(dma_cfg, 0, sizeof(*dma_cfg));
-	memset(&dma_block_cfg, 0, sizeof(dma_block_cfg));
 
```

Comment:
You can do the allocation for `dma_block_cfg` here, and there's no need to add `rfree(dma_block_cfg)` above (line 858 and 877)

Reply:
fixed, thanks!